src/safeDecoders.ts

```diff
@@ -297,6 +297,39 @@ export const object = <T extends DecoderDict>(
   return value;
 };
 
+// `record` will build a new object at runtime, and should probably not be used for large objects
+export const record = <T>(decoder: Decoder<T>): Decoder<Record<string, T>> => (
+  value,
+): Record<string, T> => {
+  // Records are plain object under the hood
+  if (!isPlainObject(value)) {
+    throw new DecodeError("record", getAccurateTypeOf(value));
+  }
+
+  const decodedValue: Record<string, T> = {};
```

Comment:
This is a bit heavy, I will create an issue to add some disambiguation between decoders, and transformers.